app/src/main/java/chat/rocket/android/authentication/login/ui/LoginFragment.kt

```diff
@@ -102,26 +124,157 @@ class LoginFragment : Fragment(), LoginView {
+        } else {
+            Log.d("Status", "nothing happening")
+        }
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        googleApiClient!!.stopAutoManage(activity!!)
+        googleApiClient!!.disconnect()
+    }
+
+    private fun buildGoogleApiClient() {
+        googleApiClient = GoogleApiClient.Builder(context!!)
+                .enableAutoManage(activity!!, {
+                    Log.d("STATUS", "ERROR: connection to client failed")
+                })
+                .addConnectionCallbacks(this)
+                .addApi(Auth.CREDENTIALS_API)
+                .build()
+    }
+
+    override fun onStart() {
+        super.onStart()
+        requestCredentials()
+    }
+
+    private fun requestCredentials() {
+        var request: CredentialRequest = CredentialRequest.Builder()
+                .setPasswordLoginSupported(true)
+                //add account types for custom login methods
+                /*.setAccountTypes(IdentityProviders.GOOGLE,IdentityProviders.FACEBOOK)*/
+                .build()
+
+        Auth.CredentialsApi.request(googleApiClient, request).setResultCallback { credentialRequestResult ->
+            val status = credentialRequestResult.status
+            if (status.isSuccess) {
+                // Auto sign-in success
+                handleCredential(credentialRequestResult.credential)
+            } else if (status.statusCode == CommonStatusCodes.RESOLUTION_REQUIRED) {
+                resolveResult(status, MULTIPLE_CREDENTIALS_READ)
+            } else if (status.statusCode == CommonStatusCodes.SIGN_IN_REQUIRED) {
+
+                //build a dialog for possible account hints
+                var hintRequest: HintRequest = HintRequest.Builder()
+                        .setHintPickerConfig(CredentialPickerConfig.Builder()
+                                .setShowCancelButton(true)
+                                .build())
+                        .setEmailAddressIdentifierSupported(true)
+                        .setAccountTypes(IdentityProviders.GOOGLE)
+                        .build()
+                var intent: PendingIntent = Auth.CredentialsApi.getHintPickerIntent(googleApiClient, hintRequest)
+                try {
+                    startIntentSenderForResult(intent.intentSender, NO_CREDENTIALS_EXIST, null, 0, 0, 0, null)
+                } catch (e: IntentSender.SendIntentException) {
+                    Log.e("STATUS", "ERROR: Could not start hint picker Intent", e);
+                }
+            } else {
+                Log.d("STATUS", "ERROR: nothing happening")
             }
         }
     }
 
+    private fun handleCredential(loginCredentials: Credential) {
+        if (loginCredentials.accountType == null) {
+            presenter.authenticateWithUserAndPassword(loginCredentials.id, loginCredentials.password!!)
+        } else if (loginCredentials.accountType == IdentityProviders.GOOGLE) {
```

Comment:
Remove the all `else if` here - we don't need it anymore.
Also, you could use `when` in this case.